coded_tools/pdf_rag.py

```diff
@@ -11,17 +11,20 @@
 #
 # END COPYRIGHT
 
+import os
 from typing import Any
 from typing import Dict
 from typing import List
 import logging
 
 from langchain_community.document_loaders import PyMuPDFLoader
 from langchain_core.documents import Document
+from langchain_core.vectorstores import VectorStore
 
 from neuro_san.interfaces.coded_tool import CodedTool
 
-from .base_rag import BaseRag
+from coded_tools.base_rag import BaseRag
+from coded_tools.base_rag import PostgresConfig
```

Comment:
^^^ should be in the file postgres_config.py

Reply:
This is back to my design comment above. There are two "axes" here --

* Persistence mechanism (in-memory, DB, other types for future?)
* Type of data being RAGged -- plain text, markdown, PDF etc.

To properly model that, you either need to be okay with multiple inheritance, or use some kind of aggregation/mix-in approach.